### PR TITLE
Added multiple tabs and integrated table display functionality in Tab 2 and added password hashing using bcrypt

### PR DESCRIPTION
**Pull request by daezel**
**File Upload Task:** Implemented multiple tab functionality in the app(File Upload Code), with separate sections for file upload, additional features, and future expansion. Added a table display feature in Tab 2 to allow users to view data from the database or other sources in a structured format, similar to the dataframe used for database display in the app.

**BugBox Solution:** Hashed passwords using bcrypt. 
Reviewed the code and verified that only admin users can view the staff information.

### CODE REVIEW

File: BugBox/game_login/db.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+''')
+
+# Function to hash a password using bcrypt
+def hash_password(password):
+    return bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
+
+# Function to update the staff table with a hashed password
+def insert_admin_user():
+    # Hash the password before inserting it into the database
+    hashed_password = hash_password('dylanBugbox')
```
Comment (ben-AI-cybersec) [inline BugBox/game_login/db.py:40]: Just need to remove this hardcoded credential as we discussed

**Comment by daezel** [comment]
Removed the credentials and used pre-hashed password as discussed.

**Comment by AmirZandiehprojects** [comment]
Hi Daezel i've also put this updated code into the scanning tool and here are my findings please address them 
Hardcoded Credentials

A pre-hashed password for the admin user 'dylan' is being directly inserted into the database.
Hardcoded credentials pose a significant security risk, as they can be easily compromised if the source code is exposed.


Insecure Password Storage

While bcrypt is being used for password hashing (which is good), the pre-hashed password is still being stored directly in the source code.
This negates the security benefits of using bcrypt and exposes the admin account to potential unauthorized access.


Admin User Creation

The code attempts to create an admin user if it doesn't exist, but then unconditionally inserts a new admin user with a pre-hashed password.
This could lead to multiple admin accounts or overwriting of existing credentials.


Database Security

The database file 'school_kids.db' is being committed to the repository.
Sensitive data, including user credentials, should never be stored in version control systems.


Potential SQL Injection Vulnerability

While the code uses parameterized queries in some places (which is good), there might be other areas in the codebase where SQL queries are constructed unsafely.
A thorough review of all database interactions is necessary to ensure protection against SQL injection attacks.


Lack of Access Controls

The code doesn't show any implementation of proper access controls or role-based permissions.
This could lead to unauthorized access to sensitive functions or data.


Insecure File Operations

The download_all_qr_codes function seems to generate and store QR codes without proper access checks or sanitization.
This could potentially be exploited to access or manipulate unauthorized data.

**Comment by daezel** [comment]
Hi @AmirZandiehprojects 

I did a few updates to my code as per your feedback. 

Added checks to ensure multiple entries of admin are not created.

Removed hardcoded credentials by using environment variables

Created a sample dotenv file so that admin can understand the structure of it and create one for himself

Removed database uploaded to git.

Ensured that qr code zip file is only accessible to admin and added checks and sanitization.

Added a small javascript code to manually refresh the page on logging out as I couldn't get experimental rerun to work(mentioned that to Ben in one the meetings that the staff login is not working as I want it to, I have to double click for the session to change even though there is a code to ensure session change which doesn't seem to work)

For remaining changes I would appreciate if you could give me a bit of more detailed info so that I can work on them.

Thankyou for your time and valuable feedback!

**Review by ben-AI-cybersec — APPROVED** [review]
Manually reviewed code. LGTM

**Pull request merged**